### PR TITLE
Shield integrity fixes and tweaks

### PR DESCRIPTION
**Pull request by RikuTheKiller**
## About The Pull Request

Fixes a bug where shield integrity messages would not display correctly.

Changes a lot of ways in which things are said on both the shield repair and shield integrity messages.

Makes shield repairs percentage based instead of a hard-coded value, since it takes far too many resources to repair a decent shield alongside taking way too long. (Engineers don't have time to just sit there for 20 seconds spamming whatever little fuel they have left on your goddamn shield. It's better to just bring spare shields at this point, which this PR remedies.) It also makes IC sense that shields of differing integrity don't take any more or less resources to repair. (It's just the hardness and toughness of the material that's in question here, not the size of the shield or anything of that nature. Repairs shouldn't change across shields.)

Shields now repair 20% of their normal integrity whenever repairs are done.

Slightly lengthens the individual repair time, though. (3 -> 4 seconds for both fumbling and repair time.)

## Why It's Good For The Game

Bugfixes are always nice, for obvious reasons.

Making the shield messages simpler, fixing some grammar mistakes and making them simply sound better adds to both immersion and quality of life.

Shields currently take far too long and far too many resources to fully repair, causing an issue where it's literally just easier to replace your shield than to repair it or bring a few spares, invalidating it entirely. This balances that. (Takes around 15 - 30 seconds depending on skill level, alongside the fact that it takes up almost an entire blowtorch.)

## Changelog
:cl:
balance: All shields now repair based on a percentage (20%)
fix: Shield integrity messages now display correctly
spellcheck: Changed wording in a lot of shield messages
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -34,21 +34,21 @@
 
 /obj/item/weapon/shield/riot/examine(mob/user, distance, infix, suffix)
 	. = ..()
-	var/health_status = (obj_integrity * 100) / (max_integrity-integrity_failure)
+	var/health_status = (obj_integrity-integrity_failure) * 100 / (max_integrity-integrity_failure)
 	if(integrity_failure && obj_integrity <= integrity_failure)
-		. += span_notice("It's broken, it won't protect anymore.")
+		. += span_notice("It's completely broken, with gaping holes everywhere!")
```
Comment (Pariah919) [inline code/game/objects/items/weapons/shields.dm:39]:
```suggestion
		. += span_notice("It's completely broken and shattered, with gaping holes all over. It won't be shielding from anything any time soon.")
```

**Review by Pariah919 — COMMENTED** [review]
How much does this repair? I can see the numbers but the average player won't know that, to clarify.

~~Anyways 25% is a nogo from me. Try 15% flat first.~~

EDIT: On further thoughts, I think going for 10+5% flat/percentage is better for the shield overall.

File: code/game/objects/items/weapons/shields.dm (hunk 1 above)
Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:39]: Why elaborate it this far? I tweaked the lines multiple times alongside bravemole and a few others in #dev-general and one of the complaints was being too verbose. This goes entirely against that.

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:39]: Also each repair of the shield takes 2 fuel + around 3 seconds to do. I could make the repair take longer, however.

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:39]: Making it 10 + 5% would literally debuff it from the current system... and it's already an issue to the point where changing shields constantly is around 3x as efficient. Fixing a shield as an engineer literally takes longer than applying a splint or two even as an untrained marine.

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:39]: Added numbers to PR description.

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:39]: Lowered integrity per repair to 20% and lengthened the repair time.

**Review by BraveMole — APPROVED** [review]

File: code/game/objects/items/weapons/shields.dm [hunk 2]
```diff
@@ -98,27 +98,27 @@
...
-		to_chat(user, span_warning("Not enough fuel to finish the task."))
+		to_chat(user, span_warning("Not enough fuel to finish the repairs."))
 		return TRUE
 
-	user.visible_message(span_notice("[user] repairs some damage on [src]."),
-	span_notice("You repair [src]."))
-	repair_damage(40)
+	user.visible_message(span_notice("[user] finishes repairing [src]."),
+	span_notice("You finish repairing [src]."))
+	repair_damage((src.max_integrity-src.integrity_failure) * 0.2)
```
Comment (QualityVan) [inline code/game/objects/items/weapons/shields.dm:121]: Bear in mind, what you're doing here is essentially spending four seconds to heal a human's extra healthpool by 60 for no resources. Scaling off the difference between max int and failure point is also kind of weird for this? And you don't need the srcs

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:121]: Repair damage is scaled off failure point since thats when it shows up as broken.

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:121]: Where do you not need the srcs? Please point that out.

Comment (BraveMole) [reply code/game/objects/items/weapons/shields.dm:121]: This line, no need of any src

Comment (RikuTheKiller) [reply code/game/objects/items/weapons/shields.dm:121]: > 

Number of the line?

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**